core/src/main/java/tc/oc/pgm/score/ScoreMatchModule.java

```diff
@@ -72,21 +76,24 @@ public void load() {
   }
 
   public boolean hasScoreLimit() {
-    return this.config.scoreLimit > 0 || hasMercyRule();
+    return scoreLimit > 0 || hasMercyRule();
   }
 
   public boolean hasMercyRule() {
     return this.mercyRule != null;
   }
 
   public int getScoreLimit() {
-    checkState(hasScoreLimit());
+    // checkState(hasScoreLimit());
```

Comment:
One does not simply comment out preconditions, they exist precisely to avoid bugs in the code, the fact that you do not pass the check means something is wrong, not that you should ignore the check.
```suggestion
    checkState(hasScoreLimit());
```